### PR TITLE
implemented 4x4 matrix multiplication using NEON intrinsics

### PR DESCRIPTION
**Pull request by gMerm**

### CODE REVIEW

**Comment by markos** [comment]
same as the other PR, each PR should be self-contained and not depend on another. You include the file but not the changes required in the Makefile in order to build it.

**Comment by gMerm** [comment]
Didn't change the B[N][N] to *B[N] because I get segmentation fault from the transposing function. Have tried other ways to fix it, but I couldn't.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (markos) [inline .vscode/settings.json]: please remove that file from the commit.

**Review by markos — APPROVED** [review]
```
Scalar matrix: 
152.00  158.00  164.00  170.00
504.00  526.00  548.00  570.00
856.00  894.00  932.00  970.00
1208.00 1262.00 1316.00 1370.00
NEON matrix: 
152.00  158.00  164.00  170.00
504.00  526.00  548.00  570.00
856.00  894.00  932.00  970.00
1208.00 1262.00 1316.00 1370.00
Scalar Matrix x Matrix: 69 sec, usec: 819371
NEON Matrix x Matrix: 18 sec, usec: 1040311
```

Again good job, approved. 
note, the algorithm can still be improved and be made even faster, without the use of addvq. we can do that as another task/PR.

**Pull request merged**